### PR TITLE
Expose `blockNumber` via filterEvents function

### LINKED ISSUE #5393: Expose block height on extrinsic results
* **I'm submitting a ...**

  - [ ] Bug report
  - [x] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

> Currently, on extrinsic result, we can get `blockHash`, `txHash` and `txIndex` but not `blockHeight`

* **What is the motivation for changing the behavior?**

> In our apps, we log the extrinsic's block height for auditing and analytics purposes. With the current setup, we would have to do another RPC call `api.rpc.chain.getBlock` to retrieve the block height. If you are happy, we can submit a PR following the same principle as https://github.com/polkadot-js/api/pull/4556 to add `blockHeight` to the extrinsic results

* **Please tell us about your environment:**

  - Version: 9.9.1
  - Environment:

    - [x] Node.js
    - [x] Browser
    - [ ] Other (limited support for other environments)

  - Language:

    - [ ] JavaScript
    - [x] TypeScript (include tsc --version)
    - [ ] Other

### PR DESCRIPTION
**Pull request by ken-centrality**
Closes #5393

### CODE REVIEW

File: packages/api/src/util/filterEvents.ts [hunk 1]
```diff
@@ -14,7 +14,8 @@ export function filterEvents (txHash: H256, { block: { extrinsics, header } }: S
           phase.isApplyExtrinsic &&
           phase.asApplyExtrinsic.eqn(txIndex)
         ),
-        txIndex
+        txIndex,
+        blockNumber: header.number.unwrap()
```
Comment (jacogr) [inline packages/api/src/util/filterEvents.ts:18]: As per the linting comments, we just need to move this up.

I'm just a bit concerned that some chains may be configured with `Compact<BlockNumber>` and others with `BlkockNumber`. So just add a bit more protection around this, here is an example used in the derives - https://github.com/polkadot-js/api/blob/f9df7a5c0df68b688b151296967599d8b0b595af/packages/api-derive/src/util/blockNumber.ts#L12-L14

Comment (ken-centrality) [reply packages/api/src/util/filterEvents.ts]: Ah I see, thank you. I've pushed a new commit with eslint issues addressed plus adding the guard.

Btw, I wasn't sure if it was a good idea to import the `api-derive/unwrapBlockNumber` function in there so I used the `isCompact` directly instead. Let me know if you want me change it around

Comment (jacogr) [reply packages/api/src/util/filterEvents.ts]: Happy as-is with the way you structured it.

**Review by jacogr — APPROVED** [review]
Thank you.

**Pull request merged**